.github/workflows/build-and-publish.yml

```diff
@@ -1,25 +1,25 @@
-name: Rust
+name: Build, Test & Publish
 
 on:
   push:
     branches: [ main ]
-  pull_request:
-    branches: [ main ]
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  build:
-
+  build-and-publish:
+  
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2
+    - name: Add musl
+      run: rustup target add x86_64-unknown-linux-musl
     - name: Build
-      run: cargo build --release --verbose
+      run: cargo build --verbose --release --target x86_64-unknown-linux-musl
```

Comment:
whats the default for this?

Reply:
the default is whatever target it's running on, so probably `x86_64-unknown-linux-gnu` for ubuntu, using `musl` is a secret hack to fully statically link the binary so it can be more portable (probably doesn't matter too much for us, but it'd let us run it in a `FROM scratch` container or on macos probably)

Reply:
https://doc.rust-lang.org/edition-guide/rust-2018/platform-and-target-support/musl-support-for-fully-static-binaries.html is a good description